### PR TITLE
Controller test cases Added

### PR DESCRIPTION
**Pull request by Harimaharajan**

### CODE REVIEW

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: maintain the endpoints as constants

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs:38]: InvalidActionRequest? what is the method testing?

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: type check already done in the above statement. Check the data?

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: How is GetAsync throwing ValidationException!?

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [reply OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: easier way would be jus taking the data and forming the url inside

File: OnlineFlightSearchAPI/Startup.cs
Comment (ghost) [inline OnlineFlightSearchAPI/Startup.cs]: need scoped?
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/dependency-injection?view=aspnetcore-2.2#service-lifetimes

**Review by ghost — CHANGES_REQUESTED** [review]
Few things missed out in the last PR since you merged it before i approved it:
Remove hardcoded data from repository classes!
I'd prefer dependency resolution inside the test methods, not at constructor level
Properly test the services, with mocked dependencies!

Do not merge with out approval!

I'd assume this PR is only for controller tests not the implementation?

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: Made the Endpoints as constants as specified

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: Unit under test is SearchController or SearchFlightDetails method?

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: could the endpoint be better?
From consumer perspective it makes no sense...

File: OnlineFlightSearchAPITestCases/Mocks/AirportRepositoryMock.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/Mocks/AirportRepositoryMock.cs]: avoid single use variables, make them inline.

File: OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs]: access modifier?

File: OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs]: where is it used?

File: OnlineFlightSearchAPITestCases/Mocks/FlightRepositoryMock.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/Mocks/FlightRepositoryMock.cs]: single use variables

File: OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs]: remove globals

File: OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs
Comment (ghost) [inline OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs]: method name could be better?

**Review by ghost — CHANGES_REQUESTED** [review]

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs]: Replaced the Method under Test

File: OnlineFlightSearchAPITestCases/Mocks/AirportRepositoryMock.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/Mocks/AirportRepositoryMock.cs]: Made the objects inline

File: OnlineFlightSearchAPITestCases/Mocks/FlightRepositoryMock.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/Mocks/FlightRepositoryMock.cs]: made the objects inline

File: OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/ControllerTests/SearchFlightsControllerTests.cs:38]: Renamed the method

File: OnlineFlightSearchAPI/Startup.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPI/Startup.cs]: Removed Dependency Injection and used Mocking

File: OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs]: Removed the Separate Mocking Class as specified

File: OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/Mocks/AirportServiceMock.cs]: Removed the Separate Mocking Class

File: OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs]: Removed Globals

File: OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs
Comment (Harimaharajan) [reply OnlineFlightSearchAPITestCases/ServiceTests/FlightServicesTests/FlightServiceTests.cs]: Removed Separate Mocking in Methods. Moved the mocking to method under test

**Comment by Harimaharajan** [comment]
Will create a new PR

**Pull request closed without merging**
